### ISSUE TITLE
possible changes to configuration files

### ISSUE THREAD

**Issue by D1dymu2** [iuvbio/pandas_ods_reader#11]
Hi iuvbio and ljnsn,

Thank you very much for writing, releasing and maintaining pandas-ods-reader.

I have installed  pandas-ods-reader 0.1.4 to a python-venv, v3.12.  It works a treat, easy to use, Well done and thank you.

What I would like to highlight is that I had pandas v2.2.2 installed to the venv.  By installing, pandas-ods-reader 0.1.4 removed the previously installed pandas v2.2.2, compiled and installed pandas v1.5.3.  

(i)	Why does pandas-ods-reader 0.1.4 require pandas v1.5.3?  
I suspect that this is because in line 21 of the file https://github.com/iuvbio/pandas_ods_reader/blob/master/pyproject.toml, the code is written pandas = "^1.5.2".  From what I understand from the site https://github.com/rust-lang/rust/issues/57443, the '^' will require a version of the same release, i.e. v1.x.x.

(ii)	Could pandas-ods-reader 0.1.4 not use pandas v2.2.2?  
Could line 21 of the .toml file be written like line 18 in the same file  `python = ">=3.8.1,<3.12"`

As an aside, in order to `pip install pandas-ods-reader 0.1.4`, _Microsoft Visual C++ 14.0 or greater is required to compile pandas v1.5.3 as part of the pandas-ods-reader install_.  
So could pandas-ods-reader be reconfigured to use the pandas library that is installed to the venv.  Ideally, could pandas-ods-reader be configured to work with libraries/modules above a certain version number [someApp >= someVersion] and remove the maximum version number and the '^' character from the .toml file.

I am willing to carry out any tests if you require.

Le gach dea-mhéin,
[translation from Gaeilge, native language of Ireland, = "with blessings and goodwill"
Tommy.

**Comment by ljnsn** [comment]
Hi @D1dymu2 , great to hear that you find the library useful.

Your request makes complete sense, indeed the library needs a few updates, I'm not sure whether bumping to pandas>2 will require further changes, but it should definitely be looked into. By now a few years later I also came to the realisation that dependency upper bounds in libraries are not a good idea, so this should also be changed :slightly_smiling_face: 

I'll look into it when I get the chance, in the meantime feel free to take a stab at it as well if you feel like, PRs are very welcome!

**Comment by D1dymu2** [comment]
Hi [ljnsn](https://github.com/ljnsn),

Thank you for the feedback.  

I would love to have a go, though I have never attempted something like this before.  I am not too sure of the process to be followed.  Do you have any pointers for a complete newbie, please.

**Comment by ljnsn** [comment]
The first thing would be to clone this repo, change the dependency spec to `>=1.5.2` and run the tests, that will already show if something breaks and if so what.

**Comment by ljnsn** [comment]
@D1dymu2 it turns out there were no issues with the bump and I also remembered that the only reason I stuck with pandas<2 before was that pandas>2 does not support Python 3.8 and I wanted to continue to support it. However, since 3.8 is EOL in September or so, I've decided to drop it now. So version 1.0.0 was released and works with pandas>2.

**Comment by D1dymu2** [comment]
Hi Iynsn,

I have downloaded and installed pandas-ods-reader 1.0.0 successfully to a
python3.10-venv on a Ubuntu 22.04.3LTS.  This pip install accepts pandas
2.2.2 as I had upgraded pandas.

Unfortunately, things did not go so well on a BASH SHELL on a Win10Pro
machine.  The install script still wants to install pandas v1.5.3, though
fails to do so as I will not install Visual Studio to this machine.

$ bash --version
GNU bash, version 4.4.23(1)-release (x86_64-pc-msys)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html
>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
(venv312)

I see something that I cannot explain...

I ran the command same command on Ubuntu server.
$ pip install pandas-ods-reader
Collecting pandas-ods-reader
  Downloading pandas_ods_reader-0.1.4-py3-none-any.whl.metadata (2.5 kB)
Collecting ezodf<0.4.0,>=0.3.2 (from pandas-ods-reader)
  Downloading ezodf-0.3.2.tar.gz (125 kB)

What am I missing?  This install attempt is trying to install
*pandas_ods_reader-0.1.4*
Why would this be?  Could it be something in my settings?

Tommy.
